odbc/src/api/data_tests.rs

```diff
@@ -339,15 +387,243 @@ mod unit {
+                            buffer_length,
+                            out_len_or_ind,
+                        )
+                    );
+                    if code == SqlReturn::SUCCESS_WITH_INFO {
+                        assert_eq!(
+                            format!(
+                                "[MongoDB][API] Buffer size \"{}\" not large enough for data",
+                                buffer_length
+                            ),
+                            format!(
+                                "{}",
+                                (*stmt_handle)
+                                    .as_statement()
+                                    .unwrap()
+                                    .read()
+                                    .unwrap()
+                                    .errors[0],
+                            ),
+                        );
+                    }
+                    if code == SqlReturn::NO_DATA {
+                        assert_eq!(0isize, *out_len_or_ind,);
+                    } else if col == ARRAY_COL {
+                        assert_eq!(
+                            "[{\"$numberInt\":\"1\"},{\"$numberInt\":\"2\"},{\"$numberInt\":\"3\"}]"
+                                .len() as isize,
+                            *out_len_or_ind
+                        );
+                    } else {
+                        assert_eq!(
+                            "你好，世界，这是一个中文句子".chars().count() as isize,
+                            *out_len_or_ind
+                        );
+                    }
+                    assert_eq!(
+                        expected.to_string(),
+                        input_wtext_to_string(char_buffer as *const _, expected.chars().count())
+                    );
+                    assert_eq!(
+                        expected.to_string(),
+                        input_wtext_to_string(char_buffer as *const _, expected.chars().count())
+                    );
+                };
+
+                str_val_test(ARRAY_COL, "[{\"$numbe", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "rInt\":\"1\"", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "},{\"$numb", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "erInt\":\"2", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "\"},{\"$num", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "berInt\":\"", SqlReturn::SUCCESS_WITH_INFO);
+                str_val_test(ARRAY_COL, "3\"}]", SqlReturn::SUCCESS);
+                str_val_test(ARRAY_COL, "", SqlReturn::NO_DATA);
+
+                str_val_test(
+                    UNICODE_COL,
+                    "你好，世界，这是一",
+                    SqlReturn::SUCCESS_WITH_INFO,
+                );
+                str_val_test(UNICODE_COL, "个中文句子", SqlReturn::SUCCESS);
+            }
+            let _ = Box::from_raw(char_buffer);
+        }
+    }
+
+    #[test]
```

Comment:
Let's add a test that validates getting the output of string data for 2 rows by pieces to confirm the var_data_indices values are being reset as expected for the next row.

Reply:
At this point, the query test would fail if this were not the case.